demo/full/scripts/controllers/ControlBar.jsx

```diff
@@ -1,87 +1,188 @@
 import React from "react";
+import { filter } from "rxjs/operators";
 import withModulesState from "../lib/withModulesState.jsx";
 import Button from "../components/Button.jsx";
 import PositionInfos from "../components/PositionInfos.jsx";
 import LivePosition from "../components/LivePosition.jsx";
 import PlayPauseButton from "./PlayPauseButton.jsx";
+import StickToLiveEdgeButton from "./StickToLiveEdgeButton.jsx";
 import FullscreenButton from "./FullScreenButton.jsx";
 import Progressbar from "./ProgressBar.jsx";
 import VolumeButton from "./VolumeButton.jsx";
 import VolumeBar from "./VolumeBar.jsx";
 
-function ControlBar({
-  player,
-  videoElement,
-  isContentLoaded,
-  isLive,
-  isStopped,
-  currentTime,
-  duration,
-  toggleSettings,
-  stopVideo,
-}) {
-  const positionElement = (() => {
-    if (!isContentLoaded) {
-      return null;
-    } else if (isLive) {
-      return <LivePosition />;
-    } else {
-      return <PositionInfos
-        position={currentTime}
-        duration={duration}
-      />;
+class ControlBar extends React.Component {
+  constructor(...args) {
+    super(...args);
+    this.state = {
+      isCatchingUp: false,
+      isStickingToTheLiveEdge: true,
+    };
+  }
+
+  componentDidMount() {
+    const { player } = this.props;
+    if (player) {
+      player.$get("loadedVideo")
+        .subscribe(() => player.dispatch("SET_PLAYBACK_RATE", 1));
```

Comment:
:warning: We never unsubscribe here!

Reply:
Also that means that the control does side effect that it only repairs if it is still enabled for the next content.

Not a natural way to think about that.

Also, we would be loading a video with the old content's playback rate, which can mean lower initial quality